src/helpers.test.js

```diff
@@ -229,36 +229,115 @@ describe('helpers', () => {
         });
-        it('should return empty object if errors are undefined', () => {
+        it('should return an empty object if errors are undefined', () => {
             const currentErrors = undefined;
             const nextErrors = undefined;
-            expect(mergeErrors(currentErrors, nextErrors)).toEqual({});
+            const expected = {};
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
+        });
+        it('should return an empty object if errors are nulls', () => {
+            const currentErrors = null;
+            const nextErrors = null;
+            const expected = {};
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
+        });
+        it('should return an empty object if errors are null and undefined respectively', () => {
+            const currentErrors = null;
+            const nextErrors = undefined;
+            const expected = {};
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
         });
-        it('should return array with errors if next error is string', () => {
-            const currentErrors = ['foo error'];
-            const nextErrors = 'bar error';
-            expect(mergeErrors(currentErrors, nextErrors)).toEqual(['foo error', 'bar error']);
+        it('should return an array with errors if next error is string and current is an array', () => {
+            const currentErrors = ['foo'];
+            const nextErrors = 'bar';
+            const expected = ['bar'];
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
         });
-        it('should return array with error if next error is string and current error is undefined', () => {
+        it('should return an array with error if next is a string and current is undefined', () => {
             const currentErrors = undefined;
-            const nextErrors = 'bar error';
-            expect(mergeErrors(currentErrors, nextErrors)).toEqual(['bar error']);
+            const nextErrors = 'bar';
+            const expected = ['bar'];
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
+        });
+        it('should return an array with error if next is null and current is a string', () => {
+            const currentErrors = 'foo';
+            const nextErrors = null;
+            const expected = ['foo'];
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
+        });
+        it('should return an array with error if next is a string and current is null', () => {
+            const currentErrors = null;
+            const nextErrors = 'bar';
+            const expected = ['bar'];
+
+            expect(mergeErrors(currentErrors, nextErrors)).toEqual(expected);
+        });
+        it('should return next errors as array if current is an array', () => {
+            const currentErrors = [];
```

Comment:
What will happend when one errors will be array of objects or array of strings and other errors will be an object?

Reply:
Added two more cases.